### ISSUE TITLE
Plan caching defeats garbage collection of cached objects

### ISSUE THREAD

**Issue by kynan** [OP2/PyOP2#382]
The plan is cached globally and keeps a reference to the `Maps` used in indirect `Args`. This defeats garbage collection of those maps, the sets they're defined on and also anything that is `ObjectCached` on those `Maps` and `Sets`.

The correct fix for this might be to make the plan `ObjectCached` on the first `Map`.

**Comment by wence-** [comment]
No longer relevant after #507.
